v2/binding/write.go

```diff
@@ -59,10 +59,45 @@ func DirectWrite(
 	return EncodingUnknown, ErrUnknownEncoding
 }
 
+// WriteEvent writes the provided event in one of the two provided writers
+// You can tweak the encoding process using the context decorators WithForceStructured, WithForceStructured, WithPreferredEventEncoding.
+// If you need to apply transformers, use ToEvent(ToMessage(e), transformers) before providing the event to this function
+// Returns:
+// * EncodingStructured, nil if message is correctly encoded in structured encoding
+// * EncodingBinary, nil if message is correctly encoded in binary encoding
+// * EncodingUnknown, ErrUnknownEncoding if message.ReadEncoding() == EncodingUnknown
+// * _, err if error happened during the encoding
+func WriteEvent(
+	ctx context.Context,
+	e *event.Event,
+	structuredWriter StructuredWriter,
+	binaryWriter BinaryWriter,
+) (Encoding, error) {
+	message := (*EventMessage)(e)
+
+	if GetOrDefaultFromCtx(ctx, preferredEventEncoding, EncodingBinary).(Encoding) == EncodingStructured {
+		if structuredWriter != nil {
```

Comment:
it bothers me that the order of the binary and structured are different.

Reply:
WDYM? it's just choosing between which of the two is preferred

Reply:
ooh I see